tests/test_association.py

```diff
@@ -0,0 +1,357 @@
+    source_locs_deg : (N, 2) list of floats
+        Positions of the DIASources to create.
+    scatter_arcsec : float
+        Scatter to add to the position of each DIASource.
+
+    Returns
+    -------
+    A lsst.afw.SourceCatalog
+    """
+    sources = afwTable.SourceCatalog(make_minimal_dia_source_schema())
+
+    if source_locs_deg is None:
+        source_locs_deg = [[idx, idx] for idx in range(n_sources)]
+
+    for src_idx in range(n_sources):
+        src = sources.addNew()
+        src['id'] = src_idx + start_id
+        coord = Coord(source_locs_deg[src_idx][0] * afwGeom.degrees,
+                      source_locs_deg[src_idx][1] * afwGeom.degrees)
+        if scatter_arcsec > 0.0:
+            coord.offset(
+                np.random.rand() * 360 * afwGeom.degrees,
+                np.random.rand() * scatter_arcsec * afwGeom.arcseconds)
+        src.setCoord(coord)
+
+    return sources
+
+
+class MeasureAssociationTestSuite(lsst.utils.tests.TestCase):
+
+    def setUp(self):
+
+        # Create an unrun AssociationTask.
+        self.assoc_task = AssociationTask()
+
+        # Create a empty butler repository and put data in it.
+        self.testDir = tempfile.mkdtemp(
+            dir=ROOT, prefix="TestAssocMeasurements-")
+        outputRepoArgs = dafPersist.RepositoryArgs(
+            root=os.path.join(self.testDir, 'repoA'),
+            mapper=obsTest.TestMapper,
+            mode='rw')
+        self.butler = dafPersist.Butler(
+            outputs=outputRepoArgs)
+        test_sources = create_test_sources(10)
+        test_dia_sources = create_test_sources(5)
+        self.butler.put(obj=test_sources,
+                        datasetType='src',
+                        dataId=dataId_dict)
+        self.butler.put(obj=test_dia_sources,
+                        datasetType='deepDiff_diaSrc',
+                        dataId=dataId_dict)
+
+        (self.tmp_file, self.db_file) = tempfile.mkstemp(
+            dir=os.path.dirname(__file__))
+        assoc_db_config = AssociationDBSqliteConfig()
+        assoc_db_config.db_name = self.db_file
+        assoc_db = AssociationDBSqliteTask(config=assoc_db_config)
+        assoc_db.create_tables()
+
+        dia_collection = DIAObjectCollection(create_test_dia_objects(5))
+        assoc_db.store(dia_collection, True)
+        assoc_db.close()
+
+    def tearDown(self):
+        del self.assoc_task
+
+        if os.path.exists(self.testDir):
+            shutil.rmtree(self.testDir)
+        if hasattr(self, "butler"):
+            del self.butler
+
+        del self.tmp_file
+        os.remove(self.db_file)
+
+    def test_valid_from_metadata(self):
+        """Verify that association information can be recovered from metadata.
+        """
+        # Insert data into the task metadata.
+        test_assoc_result = pipeBase.Struct(
+            n_updated_dia_objects=5,
+            n_new_dia_objects=6,
+            n_unassociated_dia_objects=7,)
```

Comment:
Could you set these three values as named variables first, and then reference those variables (times `u.count`) in the asserts below?